src/components/Gallery/ImageGallery.tsx

```diff
@@ -54,34 +54,55 @@ export class Gallery extends Component<GalleryProps, GalleryState> {
 
   render() {
     const { selectedAsset } = this.state;
-    if (!this.props.assets.length && !this.props.loading) {
+    const previouslySelectedSource = (
+      this.props.sdk.parameters.invocation as any
+    )?.selectedImage?.selectedSource;
+
+    if (this.props.loading) {
+      return (
+        <GalleryPlaceholder
+          handleClose={this.handleClose}
+          sdk={this.props.sdk}
+          text="Loading"
+        />
+      );
+    }
+
+    // If replacing an image
+    if (previouslySelectedSource && !this.props.assets.length) {
+      return (
+        <GalleryPlaceholder
+          handleClose={this.handleClose}
+          sdk={this.props.sdk}
+          text="Loading"
+        />
+      );
+    }
```

Comment:
Can these two conditions be collapsed into one?

Reply:
Probably. Let me play with it some more, I was having issues doing just that earlier. Kept evaluating to false despite `this.props.loading` being true.